### PR TITLE
[StructuralMechanicsApplication][MeshingApplication] Adaptative scripts

### PR DESCRIPTION
**Pull request by loumalouomega**
Requires #2612 and #2649

### CODE REVIEW

**Comment by loumalouomega** [comment]
This works now in a combined branch, hurray!!

**Comment by loumalouomega** [comment]
This can be reviewed after #2649 is merged

**Comment by loumalouomega** [comment]
Last one

**Comment by philbucher** [comment]
is there a test for this?

**Comment by loumalouomega** [comment]
Test example, this to complex to create a test. The tests are already added in form of cpp tests in the previous PR

**Comment by philbucher** [comment]
In principle I am fine, I am just afraid to break stuff if sth changes

**Comment by loumalouomega** [comment]
All the scripts are new (except mmg_process.py), so it is not breaking anything

**Comment by loumalouomega** [comment]
This is ready

**Comment by loumalouomega** [comment]
Ping

**Comment by loumalouomega** [comment]
Ping

**Comment by philbucher** [comment]
> In principle I am fine, I am just afraid to break stuff if sth changes

I meant that I am afraid if I change sth in the future I will break this since there is no test for it

**Comment by philbucher** [comment]
Btw there are some suggestions from codacy worth looking at

File: applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py]: copy paste error

File: applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py]: typo

File: applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py]: This has been discussed before but just to add my thoughts:

The processes live on the AnalysisStage and should not be in the solver.
Question: Does the remeshing-process have to be a process or could it be a utility and thus avoiding all this naming struggle?

File: applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_implicit_dynamic_solver.py
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_implicit_dynamic_solver.py]: typo

File: applications/MeshingApplication/python_scripts/mmg_process.py
Comment (philbucher) [inline applications/MeshingApplication/python_scripts/mmg_process.py:7]: why do you need to import it here?
from what I see you only use it in ExecuteFinalize => why not importing it there?

File: applications/MeshingApplication/python_scripts/mmg_process.py
Comment (philbucher) [inline applications/MeshingApplication/python_scripts/mmg_process.py]: hm I like a proper error message more

Btw if you import StructMech only here then you will get a proper error message

File: applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_analysis.py
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_analysis.py]: this is deprecated, you should also look into ["processes"]["constraints_process_list"]

File: applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py]: There is a lot of copy-pasting in this solver and the dynamic one (same for the solvers in contact) 
=> you could put some stuff into a utility or maybe use multiple inheritance (i.e. this solver would derive from `StaticMechanicalSolver` and e.g. from `BaseRemeshingSolver`)
This way you don't have to copy-paste a lot of things

**Review by philbucher — COMMENTED** [review]
global comment: Adaptive is quite an ambiguous name, it could also mean e.g. adaptive time-stepping. You don't have to change it now, but if sth else comes up we would have to think whether the names of the scripts introduced in this PR have to change to sth that has "adaptive_mesh" or similar in it

The analysis objects are rather non-standard, but I know that your application-case is also not easy to fit in this framework.

Unless someone else (@RiccardoRossi ...?) has some comments I will approve in a few days

File: applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py]: This is the only way to reduce code duplication, because otherwise you need to copy this in each solver

File: applications/MeshingApplication/python_scripts/mmg_process.py
Comment (loumalouomega) [reply applications/MeshingApplication/python_scripts/mmg_process.py:7]: For the SPRProcess

File: applications/MeshingApplication/python_scripts/mmg_process.py
Comment (loumalouomega) [reply applications/MeshingApplication/python_scripts/mmg_process.py]: I prefer to use the assert, but I raise an error

File: applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_analysis.py
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_analysis.py]: Ok

File: applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py]: OK

**Comment by loumalouomega** [comment]
I will create this common utilities

**Comment by loumalouomega** [comment]
Comments addressed @philbucher

**Comment by loumalouomega** [comment]
Ping

**Comment by philbucher** [comment]
@RiccardoRossi , I will approve on Friday unless you also want to take a look

Has been there long enough :)

**Comment by loumalouomega** [comment]
Ping

**Review by philbucher — APPROVED** [review]
gogogo :)
Sorry for the long wait...

**Comment by loumalouomega** [comment]
Finally 

imagen

**Pull request merged**
